protocol/binary.go

```diff
@@ -61,3 +67,53 @@ func (binaryProtocol) DecodeEnveloped(r io.ReaderAt) (wire.Envelope, error) {
 	e, err := reader.ReadEnveloped()
 	return e, err
 }
+
+// DecodeRequest specializes Decode and replaces DecodeEnveloped for the
```

Comment:
I’ve clarified that the request is a struct. Requests and responses are always implicit TStructs on the wire. There is nowhere in the protocol to specify otherwise.